agora/Agora/Proposal.hs

```diff
@@ -583,6 +596,15 @@ deriving via
   instance
     (PConstantDecl ProposalVotes)
 
+type PProposalEffectGroup =
+  PMap
+    'Unsorted
+    PValidatorHash
+    ( PTuple
+        PDatumHash
+        (PMaybeData (PAsData PScriptHash))
```

Comment:
I think we can encode it as `PMaybeData PScriptHash` (without the `PAsData`) right?

Reply:
No, we can't. It seems to be impossible to implement `PTryFrom PData PScriptHash` due to the `PSubtype` constraint.

Reply:
Huh, then how does [`PTxOut` do it](https://hoogle.nix.dance/file/nix/store/fb7kiqls68l24g5rj25i4qlykrl1yz8w-plutarch-lib-plutarch-1.2.0-haddock-doc/share/doc/plutarch/html/Plutarch-Api-V2-Tx.html#t:PTxOut)?

Reply:
Ah, you mean it breaks our `PTryFrom`.

Reply:
Exactly.

Reply:
I see, I guess this works for now.